app/services/exporter/service.rb

```diff
@@ -0,0 +1,64 @@
+require "zip"
+
+module Exporter
+  class Service
+    attr_reader :user
+
+    EXPORTERS = [
+      Articles,
+    ].freeze
+
+    def initialize(user)
+      @user = user
+    end
+
+    def export(send_email: false, config: {})
+      exports = {}
+
+      # export content with filenames
+      EXPORTERS.each do |exporter|
+        files = exporter.new(user).export(config.fetch(exporter.name.to_sym, {}))
+        files.each do |name, content|
+          exports[name] = content
+        end
+      end
+
+      zipped_exports = zip_exports(exports)
+
+      send_exports_by_email(zipped_exports) if send_email
+
+      update_user_export_fields
+
+      zipped_exports.rewind
```

Comment:
question, would moving `zipped_exports.rewind` to line 27 and removing line 56 yield the same result?

Reply:
No, because send_email is optional, which means that if it's called like this:

```ruby
zip_exports
rewind
send_email
```

without a final rewind the caller won't have a rewound buffer. A possible change is to put rewind at the end of the called function so that any subsequent function using the zipped buffer needs to rewind it before exiting

Something like:

```ruby
def zip_exports()
  # do stuff
  buffer.rewind
  return buffer
end

def send_email
  # send email
  buffer.rewind
end
```